x/emissions/module/rewards/topic_rewards.go

```diff
@@ -280,9 +240,17 @@ func GetAndUpdateActiveTopicWeights(
 		// If the topic is inactive, inactivate it
 		if weight.Lt(moduleParams.MinTopicWeight) {
 			err := k.InactivateTopic(ctx, topic.Id)
-			return nil, alloraMath.Dec{}, cosmosMath.Int{}, errors.Wrapf(err, "failed to inactivate topic")
+			if err != nil {
+				return nil, alloraMath.Dec{}, cosmosMath.Int{}, errors.Wrapf(err, "failed to inactivate topic")
+			}
+			continue
 		}
 
+		// Update topic active status
+		err = k.AttemptTopicReactivation(ctx, topicId)
+		if err != nil {
+			continue
+		}
```

Comment:
I think we should add a log error here, at least in the first and the last continue.